app/controllers/posts_controller.rb

```diff
@@ -37,6 +38,32 @@ def index
 
   def homepage
     @latest_post = Post.order(sticky: :desc, created_at: :desc).first
+    @preview = params[:preview] == "1" && current_user.can_administrate_livestream?
```

Comment:
This is very dangerous and will probably explode if a user is not logged in, because then `current_user` will be `nil` and calling `can_administrate_livestream?` will raise an exception. You should use nil-safe accessing here.